system/extensions/host/run-image-boot-sh.toit

```diff
@@ -89,13 +89,37 @@ for (( ; ; )); do
   rm -rf \$PREFIX/scratch
   mkdir -p \$PREFIX/scratch
 
-  # Run the image.
   export TOIT_FLASH_UUID_FILE=\$PREFIX/\$current/uuid
   export TOIT_FLASH_REGISTRY_FILE=\$PREFIX/flash-registry
-  (cd \$PREFIX/\$current; ./run-image \$PREFIX/\$current \$PREFIX/scratch)
-  exit_code=\$?
 
-  if [ \$exit_code -eq $EXIT-CODE-UPGRADE ]; then
+  run_image_pid=
+  should_exit=false
+  # Set a trap for TERM and INT signals, so we can kill the run-image child.
+  # If the run_image_pid was already set, kill it and exit immediately.
+  # Otherwise we just set the should_exit bool, and wait to get the image_pid, before
```

Comment:
```suggestion
  # Otherwise we just set the should_exit bool, and wait to get the run_image_pid, before
```

or 

```suggestion
  # Otherwise we just set the should_exit bool, and wait to get the pid, before
```

Reply:
done.